trpcultivate_phenotypes/src/Plugin/TripalImporter/TripalCultivatePhenotypesTraitsImporter.php

```diff
@@ -536,46 +650,49 @@ public function processValidationMessages($failures) {
       ]
     ];
 
-    foreach($messages as $validator_name => $default_messages) {
+    // @TODO: an alternative way to identify the validators input type.
+    $raw_row_validators = [
+      'valid_delimited_file'
+    ];
+
+    $raw_row_failed = FALSE;
+
+    foreach (array_keys($messages) as $validator_name) {
       // Check if this validator exists in the failures array, which indicates
       // that it was run.
-      if (array_key_exists($validator_name, $failures)) {
+      if (!array_key_exists($validator_name, $failures)) {
+        continue;
+      }
 
-        // ----------------------------- PASS ----------------------------------
+      // ----------------------------- PASS ----------------------------------
+      if (empty($failures[$validator_name])) {
         // Check if $failures[$validator_name] is empty, which indicates there
         // are no errors to report for this validator.
-        if (count($failures[$validator_name]) === 0 ) {
-          $messages[$validator_name] = [
-            'status' => 'pass',
-          ];
+        // If raw row validation at any point, make sure the data row validators
+        // are not set to 'pass' and remain as 'todo' since they haven't been
+        // run on every line.
```

Comment:
Maybe just a little bit more here to explain how this logic will only affect data row validators and not the metadata/file/header validators. What do you think of this?

```suggestion
        // If raw row validation fails at any point, make sure the data row 
        // validators are not set to 'pass' and remain as 'todo' since they 
        // haven't been run on every line. This approach works because the 
        // order of the validators in the default messages array ensures
        // that the raw row validators are checked for failures directly
        // before the data row validators. It also assumes that only data row
        // validators are after raw row validators in the default messages array.
```